pennylane/ops/qubit.py

```diff
@@ -348,16 +343,11 @@ class SWAP(Operation):
     num_params = 0
     num_wires = 2
     par_domain = None
-    eigvals = np.array([1, 1, 1, -1])
     matrix = np.array([[1, 0, 0, 0], [0, 0, 1, 0], [0, 1, 0, 0], [0, 0, 0, 1]])
 
     @classmethod
     def _matrix(cls, *params):
-        return SWAP.matrix
-
-    @classmethod
-    def _eigvals(cls, *params):
-        return SWAP.eigvals
+        return cls.matrix
```

Comment:
Just to confirm, anything that inherits strictly from `Operation` doesn't need to add `_eigvals` here, since it is handled via calls in the parent class (which ultimately does `np.linalg.eig(self.matrix)`. Is that correct?

Reply:
Yes. But we do it anyways if we hardcode the eigvals for performance reasons (here it's kind of "cheap" to provide the eigvals directly)